### ISSUE TITLE
Gobble not working properly

### ISSUE THREAD

**Issue by SkytAsul** [jneug/typst-codelst#18]
Hello!

Setting `gobble: false` as a parameter to `sourcecode` does not work:
```
error: expected integer, found boolean
   ┌─ @preview/codelst:2.0.1/codelst.typ:41:24
   │
41 │       line = line.slice(gobble)
   │                         ^^^^^^

help: error occurred in this call of function `codelst-gobble-blanks`
    ┌─ @preview/codelst:2.0.1/codelst.typ:288:15
    │
288 │   code-lines = codelst-gobble-blanks(code-lines, gobble)
    │                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

help: error occurred in this call of function `sourcecode`
    ┌─ @preview/codelst:2.0.1/codelst.typ:316:2
    │
316 │   sourcecode( ..args, raw(code, lang:lang, block:true))
    │   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by jneug** [comment]
`false` and `none` should disable gobbling, I guess.

For now, setting `gobble: 0` should work.

**Comment by SkytAsul** [comment]
Thanks. I have another problem: the `auto` gobble value does not work properly if used alongside `showrange`.

For instance:
````typ
#sourcecode(showrange: (5, 7))[```
    4 spaces
    4 spaces
    4 spaces
    4 spaces
        8 spaces
        8 spaces
        8 spaces
    4 spaces
```]
````
Will only remove 4 spaces instead of 8.

**Comment by jneug** [comment]
Good catch. Whitespace is removed before slicing the code to the desired range. This probably should be done the other way around.

**Comment by SkytAsul** [comment]
I don't understand why. Slicing the code to the range is done at line 152: https://github.com/jneug/typst-codelst/blob/bda9c7a6dad678749846eeeaa643cd19f30ca495/codelst.typ#L152
Whitespace removal is done at line 288: https://github.com/jneug/typst-codelst/blob/bda9c7a6dad678749846eeeaa643cd19f30ca495/codelst.typ#L288

**Comment by jneug** [comment]
The package needs to do several steps of pre-processing to the code. The line you reference is needed for parsing labels and acts on the code before syntax highlighting. The actual slicing happens around line 197, **after** highlighting (to make sure things like comments are highlighted properly, before cutting of lines).

The gobbling acts on the raw text of the code, again before highlighting.

**Comment by jneug** [comment]
Yes, that's why I implemented it the way it is. You can remove whitespace after highlighting, but it's a bit more messy. 

But I think it should be possible to simply remove the whitespace from the complete code (even lines outside the range with less whitespace), because it won't mess with syntax highlighting.

**Comment by SkytAsul** [comment]
It's a lot messier, I just finished it and it's really ugly. You can see my commit if you want: https://github.com/SkytAsul/typst-codelst/commit/2f86d1e03c6bd6c00f08330fff64be798809ce8f (I could have probably done it a little cleaner, but the main idea is here.)

If what you meant in your above message is to gobble the whitespaces even in lines that contains _less_ whitespaces that the desired gobbling amount, well that's clever but it could cause issue on languages relying on indentation? Idk if the syntax coloring is basing itself on indentation.

**Comment by jneug** [comment]
I'm currently on holidays but will habe a look at your code in the next few days.

**Comment by SkytAsul** [comment]
No hurry. It works for me so take your time, enjoy your holidays :)
